docs/DeveloperGuide.md

```diff
@@ -1,11 +1,118 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Architect class diagram](UML diagrams/OverviewClassDiagram.png)
+The Architecture Diagram above gives an overview of the different components in the application. 
+Details of the individual components are given below.  
 
+Mojo class controls the overall logic for the application. 
+It is responsible for 
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of 4 components:
+* Login: Links the user to their account.
+* Email: Stores and manages the user's email.
+* Utilities: Consists of Ui, Storage and Parser helper class. Ui interacts with the user. Storage reads data from, and writes data to the hard disk. Parser parse user's input into instructions that can be understood by the application.
+* Command: In charge of executing the user's request.
+
+For these four components,
+* The Login component exposes its functionality through the LoginController class.
+* The Email component exposes its functionality through the EmailManager class
+* The Utilities component consists of Ui, Storage and Parser classes.
+* The Command component exposes its functionality through different command classes that inherit from a base Command class.
+
+**How the architecture components interact with each other**
+![Architect sequence diagram](Sequence Diagrams/OverviewSequenceDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/24218271/113238806-148a4b00-92dc-11eb-9d43-10b73e09ee59.png)

The `providedLoginInfo` returned from LoginController - should it be a dotted arrow instead?